### PR TITLE
MP-7948 [BS-8103] NOX - Database configuration for "Weight"

### PR DESCRIPTION
**Pull request by emir-sehovic1**

### CODE REVIEW

File: src/Nox.Types.Abstractions/TypeDefinitions/NoxSimpleTypeDefinition.cs [hunk 1]
```diff
@@ -98,6 +98,9 @@ public class NoxSimpleTypeDefinition
     [IfEquals("Type", NoxType.Length)]
     public LengthTypeOptions? LengthTypeOptions { get; set; }
 
+    [IfEquals("Type", NoxType.Weight)]
+    public WeightTypeOptions? WeightTypeOptions { get; set; }
```
Comment (dejobratic) [inline src/Nox.Types.Abstractions/TypeDefinitions/NoxSimpleTypeDefinition.cs:102]: Please regenerate the schema.

File: src/Nox.Types/Types/Weight/Weight.cs [hunk 2]
```diff
@@ -1,51 +1,141 @@
...
 
 namespace Nox.Types;
 
 /// <summary>
 /// Represents a Nox <see cref="Weight"/> type and value object.
 /// </summary>
-public class Weight : Measurement<Weight, WeightUnit>
+public class Weight : ValueObject<QuantityValue, Weight>
 {
+    public const int QuantityValueDecimalPrecision = 6;
```
Comment (dejobratic) [inline src/Nox.Types/Types/Weight/Weight.cs:14]: Since the internals are visible to SqlServer project, can we mark this field as internal?

**Review by dejobratic — CHANGES_REQUESTED** [review]
Looks good. Just a few minor comments.

**Comment by emir-sehovic1** [comment]
> Looks good. Just a few minor comments.

Thanks for feedback, I've updated the implementation

**Review by dejobratic — APPROVED** [review]

File: src/Nox.Types/Nox.Types.csproj [hunk 3]
```diff
@@ -24,6 +24,7 @@
   <ItemGroup>
     <None Include="..\..\images\nox.png" Pack="true" PackagePath="" />
     <InternalsVisibleTo Include="Nox.Types.Tests" />
+	  <InternalsVisibleTo Include="Nox.EntityFramework.SqlServer" />
```
Comment (rochar) [inline src/Nox.Types/Nox.Types.csproj:27]: this should not be needed?

**Review by rochar — CHANGES_REQUESTED** [review]
Looks good, one thing to review

File: src/Nox.Types/Nox.Types.csproj (hunk 3 above)
Comment (emir-sehovic1) [reply src/Nox.Types/Nox.Types.csproj]: Property Weight.QuantityValueDecimalPrecision, and classes WeightConversion, WeightUnit were marked as internal, hence that was needed. But I've marked them now as public and removed this line as we discussed.

**Review by rochar — APPROVED** [review]

**Pull request merged**